### PR TITLE
DOCS-3604 Adds React Native Crash Reporting and Error Tracking Documentation

### PR DESCRIPTION
**Pull request by alai97**
### What does this PR do?

Single sources doc page for React Native Crash Reporting and Error Tracking, polishes up the Error Tracking section, and generates a preview for https://github.com/DataDog/dd-sdk-reactnative/pull/229.

### Motivation

DOCS-3604, RUMM-2249

### Additional Notes

Remember to revert the base branches for iOS and React Native, and update the `pull_config.yaml` file before merging!

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: local/bin/py/build/configurations/pull_config_preview.yaml [hunk 1]
```diff
@@ -775,6 +783,26 @@
...
+      - 'docs/crash_reporting.md'
+      options:
+        dest_path: '/real_user_monitoring/error_tracking/'
+        file_name: 'reactnative.md'
+        front_matters:
+          title: React Native Crash Reporting and Error Tracking
+          kind: documentation
+          beta: true
+          description: Set up Error Tracking for your React Native projects.
+          dependencies: ["https://github.com/DataDog/dd-sdk-reactnative/blob/louiszawadzki/rumm-2249/crash-reporting-docs/docs/crash_reporting.md"]
```
Comment (alai97) [inline local/bin/py/build/configurations/pull_config_preview.yaml:798]:
```suggestion
          dependencies: ["https://github.com/DataDog/dd-sdk-reactnative/blob/main/docs/crash_reporting.md"]
```

File: local/bin/py/build/configurations/pull_config_preview.yaml [hunk 2]
```diff
@@ -775,6 +783,26 @@
             - link: "real_user_monitoring/explorer/"
               tag: "Documentation"
               text: "Learn how to explore your RUM data"
+    - action: pull-and-push-file
+      branch: louiszawadzki/rumm-2249/crash-reporting-docs
```
Comment (alai97) [inline local/bin/py/build/configurations/pull_config_preview.yaml:787]:
```suggestion
      branch: main
```

File: local/bin/py/build/configurations/pull_config_preview.yaml [hunk 3]
```diff
@@ -664,25 +664,33 @@
...
       options:
-        dest_dir: '/real_user_monitoring/ios/'
-        path_to_remove: 'docs/rum_collection/'
+        dest_path: '/real_user_monitoring/error_tracking/'
+        file_name: 'ios.md'
         front_matters:
-          dependencies: [ "https://github.com/DataDog/dd-sdk-ios/blob/master/docs/rum_collection/" ]
-
+          description: "Set up Error Tracking for your iOS projects."
+          dependencies: ["https://github.com/DataDog/dd-sdk-ios/blob/master/docs/rum_collection/crash_reporting.md"]
```
Comment (alai97) [inline local/bin/py/build/configurations/pull_config_preview.yaml:676]:
```suggestion
          dependencies: ["https://github.com/DataDog/dd-sdk-ios/blob/alai97/front-matter-migration/docs/rum_collection/crash_reporting.md"]
```

File: local/bin/py/build/configurations/pull_config_preview.yaml [hunk 4]
```diff
@@ -664,25 +664,33 @@
               tag: "Documentation"
               text: "Learn how to explore your RUM data"
     
-    - action: pull-and-push-folder
+    - action: pull-and-push-file
       branch: master
```
Comment (alai97) [inline local/bin/py/build/configurations/pull_config_preview.yaml:668]:
```suggestion
      branch: alai97/front-matter-migration
```

File: local/bin/py/build/configurations/pull_config_preview.yaml [hunk 5]
```diff
@@ -664,6 +664,32 @@
...
+      - 'docs/rum_collection/crash_reporting.md'
+      options:
+        dest_path: '/real_user_monitoring/error_tracking/'
+        file_name: 'ios.md'
+        front_matters:
+          title: iOS Crash Reporting and Error Tracking
+          kind: documentation
+          beta: true
+          description: "Set up Error Tracking for your iOS projects."
+          dependencies: ["https://github.com/DataDog/dd-sdk-ios/blob/alai97/front-matter-migration/docs/rum_collection/crash_reporting.md"]
```
Comment (alai97) [inline local/bin/py/build/configurations/pull_config_preview.yaml:679]:
```suggestion
          dependencies: ["https://github.com/DataDog/dd-sdk-ios/blob/master/docs/rum_collection/crash_reporting.md"]
```

File: local/bin/py/build/configurations/pull_config_preview.yaml [hunk 6]
```diff
@@ -664,6 +664,32 @@
               tag: "Documentation"
               text: "Learn how to explore your RUM data"
     
+    - action: pull-and-push-file
+      branch: alai97/front-matter-migration
```
Comment (alai97) [inline local/bin/py/build/configurations/pull_config_preview.yaml:668]:
```suggestion
      branch: master
```

**Comment by alai97** [comment]
Preview links: 

https://docs-staging.datadoghq.com/alai97/react-native-crash-reporting-docs/real_user_monitoring/error_tracking/ios/
https://docs-staging.datadoghq.com/alai97/react-native-crash-reporting-docs/real_user_monitoring/error_tracking/reactnative/

**Review by maycmlee — APPROVED** [review]

**Pull request merged**